### PR TITLE
analog. add DataSegment command

### PR DESCRIPTION
**Pull request by Pnom**

### CODE REVIEW

File: docs/analog/commands/DataSegment.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+            <th>3</th>
+            <th>2</th>
+            <th>1</th>
+            <th>0</th>
+        </tr>
+    </thead>
+    <tbody>
+        <tr>
+            <td><a href="#last-segment">last segment</a></td>
+            <td colspan="3" align="center"><a href="#segments-number">segments number [<code>6..4</code>]</a></td>
```
Comment (DarkPark) [inline docs/analog/commands/DataSegment.md:45]:
```suggestion
            <td colspan="3" align="center"><a href="#segments-number">segments number</a></td>
```

File: docs/analog/commands/DataSegment.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+            <td>spare</td>
+            <td colspan="3" align="center"><a href="#segment-index">segment index [<code>2..0</code>]</a></td>
```
Comment (DarkPark) [inline docs/analog/commands/DataSegment.md:47]:
```suggestion
            <td colspan="3" align="center"><a href="#segment-index">segment index</a></td>
```

File: docs/analog/commands/DataSegment.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+        </tr>
+    </tbody>
+</table>
+
+#### **segmentation session id**
+
+Represents a unique identifier for a session during which segments are collected. All segments belonging to the same message share the same segmentationSessionId, allowing the collector to correlate and assemble them correctly.
```
Comment (DarkPark) [inline docs/analog/commands/DataSegment.md:54]:
```suggestion
Represents a unique identifier for a session during which segments are collected. All segments belonging to the same message share the same `segmentationSessionId`, allowing the collector to correlate and assemble them correctly.
```

File: docs/analog/commands/DataSegment.md [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+#### **last segment**
+
+The `last segment` is a boolean flag within indicates that a particular segment is the last in the sequence of segments that collectively form a complete message.
```
Comment (DarkPark) [inline docs/analog/commands/DataSegment.md:58]:
```suggestion
It is a boolean flag which indicates that a particular segment is the last in the sequence of segments that collectively form a complete message.
```

File: docs/analog/commands/DataSegment.md [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+
+| Field                   | Value                                                                  | Bits         | Hex                |
+| ----------------------- | ---------------------------------------------------------------------- | ------------ | ------------------ |
+| command id              | `30`                                                                   |              | `0x1e`             |
+| command size            | `9`                                                                    |              | `0x09`             |
+| segmentation session id | `35`                                                                   |              | `0x23`             |
+| segment flags           | last segment: `true` <br> segments number: `1` <br> segment index: `1` | `0b10010001` | `0x91`             |
+| data                    | `23 10 10 07 00 00 00`                                                 |              | `0x23101007000000` |
+
+
```
Comment (DarkPark) [inline docs/analog/commands/DataSegment.md:80]:
```suggestion
```

File: docs/analog/commands/DataSegment.md [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+| Size  | Type                                 | Field                                               |
+| ----- | ------------------------------------ | --------------------------------------------------- |
+| `1`   | `byte`                               | command id = `0x1e`                                 |
+| `1`   | `byte`                               | command size (dynamic, `2+`)                        |
+| `1`   | [Message ID](../types.md#request-id) | request/response unique identifier                  |
+| `1`   | `byte`                               | [segmentation session id](#segmentation-session-id) |
+| `1`   | `byte`                               | [segment flags](#segment-flags)                     |
+| `1..` | `bytes`                              | data. Array of bytes                                |
+
```
Comment (DarkPark) [inline docs/analog/commands/DataSegment.md:21]:
```suggestion
```

File: docs/analog/commands/DataSegment.md [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+#### **segments number**
+
+Specifies the total number of segments that make up the complete message.
+
+#### **segment index**
+
+Denotes the index or position of a particular segment within the entire message.
+
+
```
Comment (DarkPark) [inline docs/analog/commands/DataSegment.md:69]:
```suggestion
```

File: docs/analog/commands/DataSegment.md [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+## Response
+
+The response has the same format as the [request](#request).
+
+
+## See also
+
+* [hardware type](../basics.md#hardware-types)
```
Comment (DarkPark) [inline docs/analog/commands/DataSegment.md:91]: + Message ID

File: docs/analog/commands/DataSegment.md [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+### Format
+
+| Size  | Type                                 | Field                                               |
+| ----- | ------------------------------------ | --------------------------------------------------- |
+| `1`   | `byte`                               | command id = `0x1e`                                 |
+| `1`   | `byte`                               | command size (dynamic, `2+`)                        |
+| `1`   | [Message ID](../types.md#request-id) | request/response unique identifier                  |
+| `1`   | `byte`                               | [segmentation session id](#segmentation-session-id) |
+| `1`   | `byte`                               | [segment flags](#segment-flags)                     |
+| `1..` | `bytes`                              | data. Array of bytes                                |
```
[the remark this answers is not in the document]
Comment (Pnom) [reply docs/analog/commands/DataSegment.md]: fixed

File: docs/analog/commands/DataSegment.md (hunk 1 above)
Comment (Pnom) [reply docs/analog/commands/DataSegment.md]: fixed

File: docs/analog/commands/DataSegment.md (hunk 2 above)
Comment (Pnom) [reply docs/analog/commands/DataSegment.md]: fixed

File: docs/analog/commands/DataSegment.md [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+</table>
+
+#### **segmentation session id**
+
+Represents a unique identifier for a session during which segments are collected. All segments belonging to the same message share the same segmentationSessionId, allowing the collector to correlate and assemble them correctly.
+
+#### **last segment**
+
+The `last segment` is a boolean flag within indicates that a particular segment is the last in the sequence of segments that collectively form a complete message.
+
```
[the remark this answers is not in the document]
Comment (Pnom) [reply docs/analog/commands/DataSegment.md:59]: fixed

File: docs/analog/commands/DataSegment.md (hunk 4 above)
Comment (Pnom) [reply docs/analog/commands/DataSegment.md]: fixed

File: docs/analog/commands/DataSegment.md (hunk 5 above)
Comment (Pnom) [reply docs/analog/commands/DataSegment.md]: fixed

File: docs/analog/commands/DataSegment.md (hunk 7 above)
Comment (Pnom) [reply docs/analog/commands/DataSegment.md]: fixed

File: docs/analog/commands/DataSegment.md (hunk 8 above)
Comment (Pnom) [reply docs/analog/commands/DataSegment.md:88]: fixed

File: docs/analog/commands/DataSegment.md [hunk 11]
```diff
@@ -0,0 +1,89 @@
...
+
+## Response
+
+The response has the same format as the [request](#request).
+
+
+## See also
+
+* [message ID](../types.md#request-id)
```
Comment (DarkPark) [inline docs/analog/commands/DataSegment.md:87]:
```suggestion
* [Request ID](../types.md#request-id)
```
?

File: docs/analog/commands/DataSegment.md (hunk 4 above)
Comment (DarkPark) [reply docs/analog/commands/DataSegment.md]: nope )

Comment (Pnom) [reply docs/analog/commands/DataSegment.md]: did it

File: docs/analog/commands/DataSegment.md (hunk 11 above)
Comment (Pnom) [reply docs/analog/commands/DataSegment.md]: fixed
Remove `message id` from the format description. there is no such field

**Review by DarkPark — APPROVED** [review]

**Pull request merged**